File: solutions/csharp/grains/1/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n>=65) throw new ArgumentOutOfRangeException();
        ulong s = 1;
        for (int i = 1; i<n; i++) {
            s *= 2;
        }
        return s;
    }

    public static ulong Total()
    {
        ulong t = 0;
        for (int i=1; i<=64; i++) {
            t+=Square(i);
        }
        return t;
    }
}

File: solutions/lua/grains/1/grains.lua
local grains = {}

function grains.square(n)
    v=1.0
    for i=2,n,1
    do 
        v=v*2
    end
    return v
end

function grains.total()
    t=0.0
    for i=1,64,1
    do 
        t=t+grains.square(i)
    end
    return t
end

return grains
